packages/communication-react/review/stable/communication-react.api.md

```diff
@@ -475,10 +488,14 @@ export type CallingBaseSelectorProps = {
 };
 
 // @public
-export type CallingHandlers = {
+export type CallingHandlers = Omit<CallingHandlersCommon, 'onStartCall'> & {
```

Comment:
We should try very hard to avoid `Omit` types in public API because they are hard to understand for devs.

A couple questions:
* Should `onStartCall` not be included in `CallingHandlersCommon` at all? If the method is different in the two call types, it should not be included in the common interface at all. Treat interface inheritence similar to class inheritence - keep methods additive.
* If not, let's define `CallingHandlers` explicitly, and then add internal type assertions to make sure that the two interfaces agree.
* We slightly prefer `interface` over `type` in declarations.

Reply:
I think it is ok to do a little bit of duplication if it helps with readability. For the onStartCall function.

Reply:
I see the concerning here - onStartCall is the shared logic between call and teamsCall hanlders, which will be required by adapters/components. In common handlers, not return type is required, but we need to return call object for each 2 specific type otherwise it would be breaking change. I have a better way to refactor this part, lemme take a shot then update here soon